middleware/TelemetryHandler.md

```diff
@@ -2,17 +2,47 @@
 
 ## Objectives
 
-Provide a middleware component that attaches metadata to a Graph request in order help improve the SDK team improve the developer experience.
+Provide a mandatory middleware component that attaches metadata to a Graph request in order help the SDK team improve the developer experience.
```

Comment:
How are we making this mandatory?  Surely someone could just build a pipeline without it?  I know we would rather have devs include it, but I think mandatory is overstating it a bit.

Reply:
My initial thinking was to make it ubiquitous whenever a Http client is created using our client factory. We can do this by internally registering the `TelemetryHandler` as the last `DelegatingHandler` when creating a pipeline. It won't be configurable by devs, but we are the ones who will set it. Here is a [POC](https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/ad92e8ace29d5850a8f92593449da00b2a30adda/src/Microsoft.Graph.Core/Requests/GraphClientFactory.cs#L207) for this. I should rephrase it to say mandatory when created using our client factory.